client/src/transmitter/index.ts

```diff
@@ -89,88 +127,116 @@ function requested_transport() {
+  const is_h264_mime_type = cast.wsMimeType.indexOf('h264') > 0
 
-  const transport =
-    requested_transport() ??
-    (cast.wsMimeType.indexOf('h264') > 0 ? 'ws' : 'wrtc')
-  const connectTime = Date.now()
-  let newCast: CastSession
+  const transport = requested_transport ?? (is_h264_mime_type ? 'ws' : 'wrtc')
+
+  const connect_time = Date.now()
+
+  let new_cast: CastSession
   if (transport === 'wrtc') {
-    newCast = cast.viaWebRTC(stream, _streamKey)
+    new_cast = cast.viaWebRTC(stream, _stream_key)
   } else {
-    newCast = cast.viaWebSocket(stream, _streamKey, !_playbackId)
+    new_cast = cast.viaWebSocket(stream, _stream_key, !_playback_id)
   }
-  curr_cast = newCast
 
-  newCast.onError = (isTransient) => {
-    if (curr_cast !== newCast) {
+  _curr_cast = new_cast
+
+  new_cast.onError = (isTransient) => {
+    if (_curr_cast !== new_cast) {
       return
     }
-    stop_recording()
-    curr_cast = null
+    stopRecording()
+    _curr_cast = null
 
-    const connectionAge = Date.now() - connectTime
-    const shouldRetry = isTransient && connectionAge >= minRetryThreshold
-    if (shouldRetry) {
-      start_recording(stream)
+    const connection_age = Date.now() - connect_time
+    const should_retry = isTransient && connection_age >= MIN_RETRY_THRESHOLD
+    if (should_retry) {
+      startRecording(stream)
     }
   }
 
-  record.style.background = '#dd0000'
-  record.style.borderColor = '#dd0000'
+  _record.style.background = '#dd0000'
+  _record.style.borderColor = '#dd0000'
 
-  video.style.opacity = '1'
+  _video.style.opacity = '1'
 
-  if (_playbackId) playbackUrl.classList.add('visible')
+  if (_playback_id) {
+    _playback_url.classList.add('visible')
+  }
 
-  record_flash_interval = setInterval(() => {
-    record_frash_dim = !record_frash_dim
+  _record_flash_interval = setInterval(() => {
+    _record_frash_dim = !_record_frash_dim
 
-    if (record_frash_dim) {
-      record.style.opacity = '0'
+    if (_record_frash_dim) {
+      _record.style.opacity = '0'
     } else {
-      record.style.opacity = '1'
+      _record.style.opacity = '1'
     }
   }, 1000)
 }
 
-let record_flash_interval
-
-function stop_recording() {
-  if (!curr_cast) {
+function stopRecording() {
+  if (!_curr_cast) {
     return
   }
-  console.log('stop_recording')
+  console.log('stopRecording')
 
-  curr_cast.close()
-  curr_cast = null
+  _curr_cast.close()
+  _curr_cast = null
 
-  record.style.opacity = '1'
-  record.style.background = '#dddddd'
-  record.style.borderColor = '#dddddd'
+  _record.style.opacity = '1'
+  _record.style.background = '#dddddd'
+  _record.style.borderColor = '#dddddd'
 
-  video.style.opacity = '0.5'
+  _video.style.opacity = '0.5'
 
-  clearInterval(record_flash_interval)
+  clearInterval(_record_flash_interval)
 }
 
-function refresh_recording(stream: MediaStream): void {
-  if (curr_cast) {
-    stop_recording()
-    start_recording(stream)
+async function setMicrophoneStream(): Promise<void> {
```

Comment:
**nit:** How about `getCanvasAndMicStream` and callers use it like `_stream = await getCanvasAndMicStream()`? 

With the current name, I'd expected a function that actually receives a `MediaStream` object and does only the part of the microphone tracks, nothing related to the canvas.

Reply:
I just changed it to `setupMicrophone` and called it a day.